apps/server/src/shared/repo/task/task.repo.integration.spec.ts

```diff
@@ -286,143 +386,104 @@ describe('TaskRepo', () => {
-				em.clear();
-
-				// unset private using a raw query because it's impossible using the ORM
-				await em
-					.getDriver()
-					.getConnection()
-					.getCollection('homeworks')
-					.updateOne({ _id: task._id }, { $unset: { private: '' } });
-
-				const [result, total] = await repo.findAllByParentIds({ creatorId: teacher.id }, { draft: false });
-				expect(total).toEqual(1);
-				expect(result[0].id).toEqual(task.id);
-			});
-
-			it('should return tasks with both status when no filter is applied', async () => {
-				const teacher = userFactory.build();
-				const task1 = taskFactory.draft().build({ creator: teacher });
-				const task2 = taskFactory.build({ creator: teacher });
-
-				await em.persistAndFlush([task1, task2]);
-				em.clear();
-
-				const [result, total] = await repo.findAllByParentIds({ creatorId: teacher.id });
-				expect(total).toEqual(2);
-				const taskNames = result.map((o) => o.name);
-				expect(taskNames.includes(task1.name)).toBe(true);
-				expect(taskNames.includes(task2.name)).toBe(true);
-			});
-
-			it('should filter tasks by dueDate after a given date', async () => {
-				const teacher = userFactory.build();
-				const threeWeeksinMilliseconds = 1.814e9;
-				const dueDate1 = new Date(Date.now() - threeWeeksinMilliseconds);
-				const dueDate2 = new Date();
-				const task1 = taskFactory.build({ creator: teacher, dueDate: dueDate1 });
-				const task2 = taskFactory.build({ creator: teacher, dueDate: dueDate2 });
-
-				await em.persistAndFlush([task1, task2]);
-				em.clear();
+			describe('when due date filter is applied', () => {
+				it('should filter tasks by dueDate after a given date', async () => {
+					const teacher = userFactory.build();
+					const threeWeeksinMilliseconds = 1.814e9;
+					const dueDate1 = new Date(Date.now() - threeWeeksinMilliseconds);
+					const dueDate2 = new Date();
+					const task1 = taskFactory.build({ creator: teacher, dueDate: dueDate1 });
+					const task2 = taskFactory.build({ creator: teacher, dueDate: dueDate2 });
+
+					await em.persistAndFlush([task1, task2]);
+					em.clear();
+
+					const oneWeekInMilliseconds = 6.048e8;
+
+					const [result, total] = await repo.findAllByParentIds(
+						{ creatorId: teacher.id },
+						{ afterDueDateOrNone: new Date(Date.now() - oneWeekInMilliseconds) }
+					);
+					expect(total).toEqual(1);
+					expect(result[0].id).toEqual(task2.id);
+				});
 
-				const oneWeekInMilliseconds = 6.048e8;
+				it('should return tasks if they have no dueDate', async () => {
+					const teacher = userFactory.build();
+					const threeWeeksinMilliseconds = 1.814e9;
+					const dueDate1 = new Date(Date.now() - threeWeeksinMilliseconds);
+					const dueDate2 = new Date();
+					const task1 = taskFactory.build({ creator: teacher, dueDate: dueDate1 });
+					const task2 = taskFactory.build({ creator: teacher, dueDate: dueDate2 });
+					const task3 = taskFactory.build({ creator: teacher, dueDate: undefined });
```

Comment:
recommended using null instead of undefined, right?

Reply:
why is this recommended? the property `dueDate` is optional in the `TaskEntity` and `ITaskProperties`. which means `Date | undefined`. so `null` might not be the right choice.

Reply:
sorry my bad